### PR TITLE
Add fish shell completion for bazel

### PR DESCRIPTION
**Pull request by akirabaruah**
Addresses issue #5088 

Test: Follow steps in README.md and use bazel in fish shell.

### CODE REVIEW

File: scripts/fish/completions/bazel.fish [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+end
+
+function __bazel_get_subcommand_args \
+    -d 'Print an argument string for subcommand completion'
+    set -l subcommand $argv[1]
+    set -l arg_type (__bazel_get_subcommand_arg_type $subcommand)
+
+    switch $arg_type
+        case "label"
+            echo "($__bazel_command query '//...' 2>/dev/null)"
```
Comment (meisterT) [inline scripts/fish/completions/bazel.fish:94]: I assume we should add `-k` to all these queries to return results even when you can't do `bazel query ...`.

Comment (akirabaruah) [reply scripts/fish/completions/bazel.fish]: Yep, thanks for pointing that out. Done!

**Comment by akirabaruah** [comment]
> Please include the default license header.

Done

> Also, you probably want to do the equivalent of this in the installer?
> [74c274a](https://github.com/bazelbuild/bazel/commit/74c274a4ed8f97ab32724641bb011cee2e057a76)

Done

> Even after the change that I proposed it doesn't complete for me in the Bazel repository for `bazel test //src/test/<TAB><TAB>`.
> 
> Can you check what's going on there?

Hmm, just to sanity check:
1. Does `bazel <tab><tab>` complete subcommands for you like so?
Peek 2020-05-28 22-41

2. Does `bazel query -k 'tests(//...)' 2>/dev/null` list targets as expected?

`bazel test //src/test/<tab><tab>` works for me locally, but it does take a few seconds the first time (as demonstrated in the screencap above). The subcommand completions are derived by running `bazel help`, which is why it takes so long. Could it be that your shell was hanging on running bazel in the background?

If needed, I think we can speed it up by building fish.bazel using templates, similar to bash completion.

**Comment by meisterT** [comment]
I just tried again, it works now. Perhaps I was too impatient.

I think there's still room for improvement, e.g. if one does starts the completion with the prefix `//foo/bar/zoo` instead of querying `//...` you could query `//foo/bar/...`.
Another idea: if it's possible to display extra information then it would be nice to query with `--output=label_kind` and display the full info.

But this is a great start and I am going to merge it as is. Thanks for your contribution!

**Comment by akirabaruah** [comment]
Awesome. Glad it works! I agree that there's much to be improved, so I'm happy to follow up with some more PRs to address your suggestions.

**Comment by gorakhargosh** [comment]
An improvement for this would be if I could use https://github.com/junegunn/fzf to select the target. Consider installing fzf and run the following command.

`bazel test (bazel query '...' --output label | fzf)`

*mind blown*

**Comment by meisterT** [comment]
@gorakhargosh That's unrelated to fish and works in any shell, right?

**Comment by gorakhargosh** [comment]
@meisterT sure, `fzf` can be used with any shell. The command I've used uses fish shell expansion however. It may make a good addition to the autocompletion feature set. The equivalent in bash would be:

`bazel test $(bazel query '...' --output label | fzf)`

**Pull request closed without merging**